octoprint_octorelay/__init__.py

```diff
@@ -262,6 +263,9 @@ def reducer(agg, task):
         )
 
     def update_ui(self):
+        if self.ui_update_lock:
+            return
```

Comment:
i don't like this approach, since the next call of this method can be caused by the most recent data state, while the payload for ongoing execution may already have been shaped.
that can lead to missing the important updates.
what is really required is an actual thread lock that awaits to be released.